Update documentation around `next/cache` (#289)
cloudflare/next-on-pages: docs/supported.md

Incremental Static Regeneration (ISR) is a rendering mode in Next.js that allows you to automatically cache and periodically regenerate pages with fresh data. Next.js [does not support](https://nextjs.org/docs/pages/building-your-application/rendering/incremental-static-regeneration) building ISR pages for the edge runtime, and as such, pages should be changed to use server side rendering (SSR) instead.

> ISR pages are built by the Vercel CLI to generate Vercel Serverless Functions. These are Node.js Lambda functions that can be called in the background while serving the page from the cache. It is not possible to use these with Cloudflare Pages and they are not compatible with the [edge runtime](https://nextjs.org/docs/app/api-reference/edge) currently.

Reviewer: I'm not sure I agree with this change. I specifically referred to them as prerender functions here because there are two types of serverless functions, and ISR uses the prerender variant.
- [Prerender functions](https://vercel.com/docs/build-output-api/v3/primitives#prerender-functions) that come with a prerender config and fallback file.
- [Serverless functions](https://vercel.com/docs/build-output-api/v3/primitives#serverless-functions) that are always run.

Another distinction relevant to this library is that in the case of prerender functions, we use the fallback files, but with regular serverless functions, we error during the build.

Perhaps it would be better if we called them Prerender functions but included a link to the Vercel docs on them?

```suggestion
ISR pages are built by the Vercel CLI to generate Vercel [Prerender Functions](https://vercel.com/docs/build-output-api/v3/primitives#prerender-functions). These are Node.js Lambda functions that can be called in the background while serving the page from the cache. It is not possible to use these with Cloudflare Pages and they are not compatible with the [edge runtime](https://nextjs.org/docs/app/api-reference/edge) currently.
```

### Revalidating Data and `next/cache`

Revalidation and `next/cache` are not supported on Cloudflare Pages. This is used by the default `fetch` cache, which forms part of the incremental cache for revalidating data inside the App Router. Revalidating tags and data for an entire path also uses `next/cache`.

> Because of this, `next/cache` functionality currently is not supported on Cloudflare Pages. The Next.js cache does work when self-hosting, and can be optionally provided a [custom cache handler](https://nextjs.org/docs/app/api-reference/next-config-js/incrementalCacheHandlerPath). It's possible this could use Cloudflare KV or Durable Objects in the future.

Reviewer 2: The previous paragraph mentions that `next/cache` is not supported on Cloudflare Pages, and here in this paragraph we say "Because of this, `next/cache` functionality currently is not supported on Cloudflare Pages."
That looks a bit incorrect and redundant to me. 🤔 

```suggestion
The Next.js cache does however work when self-hosting by optionally providing a [custom cache handler](https://nextjs.org/docs/app/api-reference/next-config-js/incrementalCacheHandlerPath). It's possible this could use Cloudflare KV or Durable Objects in the future.
```
